### PR TITLE
Additional tests for the top-level 'cel' package

### PR DESCRIPTION
**Pull request by TristonianJones**
Additional tests for the top-level `cel` package, including:

* Custom Macros
* Custom Type Provider
* IO tests

### CODE REVIEW

**Comment by TristonianJones** [comment]
@tobiaszheller I think this should get the top-level package up to 83% coverage which will satisfy the awesome-go quality standards: https://github.com/avelino/awesome-go/pull/2756

File: cel/env.go [hunk 1]
```diff
@@ -204,12 +204,8 @@ type env struct {
 
 // Check implements the Env interface method.
 func (e *env) Check(ast Ast) (Ast, Issues) {
-	pe, err := AstToParsedExpr(ast)
-	if err != nil {
-		errs := common.NewErrors(ast.Source())
-		errs.ReportError(common.NoLocation, err.Error())
-		return nil, &issues{errs: errs}
-	}
+	// Note, errors aren't currently possible on the Ast to ParsedExpr conversion.
```
Comment (gabrielsz) [inline cel/env.go:207]: If this is expected to change ~soon, let's link the github issue for it. If it's a note about a design limitation, it should be OK to leave as-is.

File: cel/io_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	}
+	checked, err := AstToCheckedExpr(ast)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ast3 := CheckedExprToAst(checked)
+	if !proto.Equal(ast3.Expr(), ast.Expr()) {
+		t.Fatalf("Got %v, wanted %v", ast3, ast)
+	}
+
```
Comment (gabrielsz) [inline cel/io_test.go]: Nit: Extra \n

File: cel/io_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+func TestIO_AstToString(t *testing.T) {
+	stdEnv, _ := NewEnv()
+	in := "a + b - (c ? (-d + 4) : e)"
+	ast, _ := stdEnv.Parse(in)
```
Comment (gabrielsz) [inline cel/io_test.go:61]: This is assuming all of the strings are parse-able, right?

**Review by gabrielsz — APPROVED** [review]
LGTM. I've left low priority nitpicks that don't need a re-review

File: cel/env.go (hunk 1 above)
Comment (TristonianJones) [reply cel/env.go:207]: Yep, this is just a design note. For the foreseeable future, it's not possible to produce an error when converting between an `Ast` and a `ParsedExpr`, but it is possible with a `CheckedExpr`. In order to keep the interface symmetric there's an `error` on the return.

File: cel/io_test.go (hunk 3 above)
Comment (TristonianJones) [reply cel/io_test.go:61]: That's right, the test is assuming the string will parse since the purpose of the test is to show the serialization.

**Pull request merged**